tlatools/src/tlc2/tool/SimulationWorker.java

```diff
@@ -68,11 +68,12 @@
 	// no trace files are generated.
 	private final String traceFile;
 	
-	// A counter that tracks the number of generated states. This counter may be
-	// shared among workers, so its count may be greater than the number of states
+	// A counter that tracks the number of generated states/traces. This counter may be
+	// shared among workers, so its count may be greater than the number of states/traces
 	// generated by this worker alone. It is the duty of this worker, though, to
-	// update it whenever it generates a new state.
+	// update it whenever it generates a new state or trace.
 	private final LongAdder numOfGenStates;
+	private final LongAdder numOfGenTraces;
```

Comment:
Note: If we think it's useful enough, we could include this in the output of the Progress Reporter, in the form of average total traces generated per second.

Reply:
Do you think traces/sec is useful for end-users?

Reply:
I personally think that trace generation count is an interesting and useful metric for simulation mode, since generation of a trace is the relatively logical "unit of work" for each simulation worker. I'm fine with leaving it out of reporting for now, though. Are you ok if we leave the counter there if we want to add it to the metrics output later?